apps/src/StudioApp.js

```diff
@@ -2121,6 +2123,9 @@ StudioApp.prototype.configureDom = function (config) {
   var container = document.getElementById(config.containerId);
   var codeWorkspace = container.querySelector('#codeWorkspace');
 
+  const isSignedOut = !config.isSignedIn;
+  const isStandaloneProject = config.scriptId === undefined;
```

Comment:
Nit: could this be `!config.scriptId`? Or are we looking for when it is `undefined` but not `null`

Reply:
This one was initially my suggestion for clarity, but I'm happy either way Kelby esp considering how the 'isSignedOut' is now formatted above it!

Reply:
Hannah and I were seeing `undefined` in the console so we opted to use this to be extra safe since neither of us is super familiar with this part of the codebase. Def happy to update though if others think `null` is a better choice!

Reply:
> This one was initially my suggestion for clarity, but I'm happy either way Kelby esp considering how the 'isSignedOut' is now formatted above it!

@hannahbergam thanks to failing tests, I discovered this was already a const in the currentUserRedux 🤩 https://github.com/code-dot-org/code-dot-org/blob/46de3d1c1794fd432a545cfa353600c64ae270a8/apps/src/templates/currentUserRedux.js#L280-L282

Reply:
Sounds good! I'm good with either then!

Reply:
I'd agree with Turner on using the `!` here, but I'm interested in Sanchit's thoughts as well! Totally understand the cautiousness

Reply:
Oof yeah this is a tricky one 😅 I agree that `!` is probably the right move given that these values come from the server and I don't know for certain if missing values are assigned `null` or `undefined` (or even blank for that matter).

That being said - if we're trying to determine whether this is a standalone project, I _think_ the check we want is `config.level.isProjectLevel`. I believe we could have an unassigned script ID on any `levels/<level ID>` pages too, which aren't typically user-facing but still aren't standalone projects.

Reply:
Re: using the `isSignedIn` selector in `MusicView` - I think what you have is totally fine, but if you do want to use that function, I'd use in in the `connect` redux function at the bottom of the file to expose `isSignedIn` as a prop to the component, e.g.

```
const MusicView = connect(
  state => ({
    currentLevelId: state.progress.currentLevelId,
    ...,
>   isSignedIn: isSignedIn(state.currentUser),
    ...,
    )},
    ...
)(UnconnectedMusicView);
```
And then you can access `this.props.isSignedIn` as a prop in the component

Reply:
> if we're trying to determine whether this is a standalone project, I think the check we want is `config.level.isProjectLevel`

I tried this, but it's returning `true` on a standalone project too. I'll leave this as-is for now, and we can keep an eye on it once it goes live.

Reply:
Right, yep it should return `true` for standalone projects and `false`/`undefined` for non-standalone projects like script levels and `level/level_id` pages.

Reply:
Oh got it! (The name threw me off haha.) In that case, it works! 🎉